src/garage/torch/algos/multi_task_sac.py

```diff
@@ -0,0 +1,246 @@
+            reward_scale=1.0,
+            optimizer=torch.optim.Adam,
+            steps_per_epoch=1,
+            num_evaluation_trajs=5,
+    ):
+
+        super().__init__(
+            policy=policy,
+            qf1=qf1,
+            qf2=qf2,
+            replay_buffer=replay_buffer,
+            env_spec=env_spec,
+            max_path_length=max_path_length,
+            gradient_steps_per_itr=gradient_steps_per_itr,
+            use_automatic_entropy_tuning=use_automatic_entropy_tuning,
+            alpha=alpha,
+            target_entropy=target_entropy,
+            initial_log_entropy=initial_log_entropy,
+            discount=discount,
+            buffer_batch_size=buffer_batch_size,
+            min_buffer_size=min_buffer_size,
+            target_update_tau=target_update_tau,
+            policy_lr=policy_lr,
+            qf_lr=qf_lr,
+            reward_scale=reward_scale,
+            optimizer=optimizer,
+            steps_per_epoch=steps_per_epoch,
+            num_evaluation_trajs=num_evaluation_trajs,
+            eval_env=eval_envs)
+        self._num_tasks = num_tasks
+        self._eval_envs = eval_envs
+        if self.use_automatic_entropy_tuning and not alpha:
+            if target_entropy:
+                self.target_entropy = target_entropy
+            else:
+                self.target_entropy = -np.prod(
+                    self.env_spec.action_space.shape).item()
+            self.log_alpha = torch.Tensor([self._initial_log_entropy] *
+                                          self._num_tasks).requires_grad_()
+            self.alpha_optimizer = optimizer([self.log_alpha] *
+                                             self._num_tasks,
+                                             lr=self.policy_lr)
+        else:
+            self.log_alpha = torch.Tensor([alpha]).log()
+        self._epoch_mean_success_rate = []
+        self._epoch_median_success_rate = []
+
+    def _get_log_alpha(self, **kwargs):
+        """Return the value of log_alpha.
+
+        This function exists in case there are versions of MTSAC that need
+        access to a modified log_alpha, such as multi_task MTSAC.
+
+        Args:
+            kwargs(dict): keyword args that can be used in retrieving the
+                log_alpha parameter. Unused here.
+
+        Returns:
+            torch.Tensor: log_alpha
+
+        """
+        obs = kwargs['obs']
+        log_alpha = self.log_alpha
+        one_hots = obs[:, :self._num_tasks]
+        ret = torch.mm(one_hots, log_alpha.unsqueeze(0).t()).squeeze()
+        assert ret.size() == torch.Size([self.buffer_batch_size])
+        return ret
+
+    def optimize_policy(self, itr, samples_data):
+        """Optimize the policy q_functions, and temperature coefficient.
+
+        Args:
+            itr (int): Iterations.
+            samples_data (list): Processed batch data.
+
+        Returns:
+            torch.Tensor: loss from actor/policy network after optimization.
```

Comment:
please provide shape information for all tensor inputs/outputs on public functions